sql/flow.go

```diff
@@ -0,0 +1,161 @@
+	"github.com/docker/docker/pkg/archive"
+	specs "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+type MountVolume struct {
+	SourceDirectory string
+	TargetDirectory string
+}
+
+const (
+	SQLCliDockerfilePath      = ".Dockerfile.sql_cli"
+	SQLCLIDockerfileWriteMode = 0o600
+	SQLCliDockerImageName     = "sql_cli"
+	PythonVersion             = "3.9"
+)
+
+type DockerBinder struct {
+	cli *client.Client
+}
+
+type DockerBind interface {
+	ImageBuild(ctx context.Context, buildContext io.Reader, options *types.ImageBuildOptions) (types.ImageBuildResponse, error)
+	ContainerCreate(ctx context.Context, config *container.Config, hostConfig *container.HostConfig, networkingConfig *network.NetworkingConfig, platform *specs.Platform, containerName string) (container.ContainerCreateCreatedBody, error)
+	ContainerStart(ctx context.Context, containerID string, options types.ContainerStartOptions) error
+	ContainerWait(ctx context.Context, containerID string, condition container.WaitCondition) (<-chan container.ContainerWaitOKBody, <-chan error)
+	ContainerLogs(ctx context.Context, container string, options types.ContainerLogsOptions) (io.ReadCloser, error)
+}
+
+func (d DockerBinder) ImageBuild(ctx context.Context, buildContext io.Reader, options *types.ImageBuildOptions) (types.ImageBuildResponse, error) {
+	return d.cli.ImageBuild(ctx, buildContext, *options)
+}
+
+func (d DockerBinder) ContainerCreate(ctx context.Context, config *container.Config, hostConfig *container.HostConfig, networkingConfig *network.NetworkingConfig, platform *specs.Platform, containerName string) (container.ContainerCreateCreatedBody, error) {
+	return d.cli.ContainerCreate(ctx, config, hostConfig, networkingConfig, platform, containerName)
+}
+
+func (d DockerBinder) ContainerStart(ctx context.Context, containerID string, options types.ContainerStartOptions) error {
+	return d.cli.ContainerStart(ctx, containerID, options)
+}
+
+func (d DockerBinder) ContainerWait(ctx context.Context, containerID string, condition container.WaitCondition) (body <-chan container.ContainerWaitOKBody, err <-chan error) {
+	return d.cli.ContainerWait(ctx, containerID, condition)
+}
+
+func (d DockerBinder) ContainerLogs(ctx context.Context, containerID string, options types.ContainerLogsOptions) (io.ReadCloser, error) {
+	return d.cli.ContainerLogs(ctx, containerID, options)
+}
+
+var newDockerClient = func() (DockerBind, error) {
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	if err != nil {
+		return nil, err
+	}
+	return &DockerBinder{cli: cli}, nil
+}
+
+func getContext(filePath string) io.Reader {
+	ctx, _ := archive.TarWithOptions(filePath, &archive.TarOptions{})
+	return ctx
+}
+
+func CommonDockerUtil(cmd, args []string, flags map[string]string, mountDirs []string) {
+	ctx := context.Background()
+	cli, err := newDockerClient()
+	if err != nil {
+		err = fmt.Errorf("docker client initialisation failed for command %v: %w", cmd, err)
```

Comment:
fix typo for initialization

Reply:
That's the British spelling - not necessarily a typo :see_no_evil: Do we use American spelling in the Astro CLI?

Reply:
I was wondering about that British / American too... :)

Reply:
Yeah, American English is what we use 😅